src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -90,15 +100,17 @@ export class DefaultTelemetryService implements TelemetryService {
     }
 
     public record(event: TelemetryEvent): void {
-        if (this.telemetryEnabled) {
-            this._eventQueue.push(event)
-        }
+        this._eventQueue.push(event)
```

Comment:
We should only capture while enabled. Essentially everything should stop if disabled, not just the flush.

Reply:
This does mean that we will lose events while the user has not expressed a preference, but I am fine with making this change if you feel strongly about it.

Reply:
Let's discuss with the team. Mitigation would be to log an onTelemetryEnabled event, at least to signal "hello, we're sending data now!"

Reply:
Resolving this by breaking this into 3 states:

Telemetry enabled:
* capture ok
* publish ok

Telemetry disabled:
* no capture
* no publish

Telemetry dialog not dismissed
* capture ok
* no publish